### ISSUE TITLE
kube-proxy pod unable to write to /sys with cri-o

### ISSUE THREAD

**Issue by matthias50** [cri-o/cri-o#2625]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
**Description**

When I start kube-proxy using cri-o I get this error from kube-proxy:

```
E0717 20:59:57.644545       1 conntrack.go:127] sysfs is not writable: {Device:sysfs Path:/sys Type:sysfs Opts:[ro seclabel nosuid nodev noexec relatime] Freq:0 Pass:0} (mount options are [ro seclabel nosuid nodev noexec relatime])
```

The same pod manifest works with Docker (i.e. no such errors)

**Steps to reproduce the issue:**
1. start kube-proxy using a static pod manifest.

**Describe the results you received:**
kube-proxy can't write to /sys

**Describe the results you expected:**
kube-proxy is able to write to /sys since it is a privileged container.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
crio version 1.13.6-1.dev.rhaos4.1.gitee2e748.el7-dev
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Running in AWS.

kube-proxy pod spec:

```
apiVersion: v1
kind: Pod
metadata:
 annotations:
   scheduler.alpha.kubernetes.io/critical-pod: ""
 creationTimestamp: null
 labels:
   k8s-app: kube-proxy
   tier: node
 name: kube-proxy
 namespace: kube-system
spec:
 priorityClassName: "system-node-critical"
 containers:
 - command:
   - /bin/sh
   - -c
   - exec kube-proxy
     --config=/srv/kubernetes/kube-proxy/kube-proxy-config.yaml
     2>&1
   image: <redacted>.dkr.ecr.us-west-2.amazonaws.com/mirror/kube-proxy:v1.13.7
   imagePullPolicy: IfNotPresent
   name: kube-proxy
   resources:
     requests:
       cpu: 100m
   securityContext:
     privileged: true
   volumeMounts:
   - mountPath: /srv/kubernetes/kube-proxy
     name: kube-proxy
     readOnly: true
   - mountPath: /etc/ssl/certs
     name: ssl-certs-hosts
     readOnly: true
   - mountPath: /run/xtables.lock
     name: xtables-lock
     readOnly: false
   - mountPath: /lib/modules
     name: lib-modules
     readOnly: false
 hostNetwork: true
 volumes:
 - hostPath:
     path: /srv/kubernetes/kube-proxy
   name: kube-proxy
 - hostPath:
     path: /run/xtables.lock
   name: xtables-lock
 - hostPath:
     path: /usr/share/ca-certificates
   name: ssl-certs-hosts
 - hostPath:
     path: /lib/modules
   name: lib-modules
status: {}
```

```
# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

```
# uname -a
Linux ip-10-121-1-143.us-west-2.compute.internal 5.2.1-1.el7.elrepo.x86_64 #1 SMP Sun Jul 14 08:15:04 EDT 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by rhatdan** [comment]
Do we switch all mount points to read/write when running in --privileged mode in CRI-O?  We should.

**Comment by rhatdan** [comment]
```
# podman run fedora mount | grep \(ro
sysfs on /sys type sysfs (ro,relatime,seclabel)
cgroup on /sys/fs/cgroup type tmpfs (ro,relatime,context="system_u:object_r:container_file_t:s0:c63,c525",size=1024k)
cgroup on /sys/fs/cgroup/blkio type cgroup (ro,relatime,seclabel,blkio)
cgroup on /sys/fs/cgroup/pids type cgroup (ro,relatime,seclabel,pids)
cgroup on /sys/fs/cgroup/freezer type cgroup (ro,relatime,seclabel,freezer)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (ro,relatime,seclabel,hugetlb)
cgroup on /sys/fs/cgroup/cpuset type cgroup (ro,relatime,seclabel,cpuset)
cgroup on /sys/fs/cgroup/memory type cgroup (ro,relatime,seclabel,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (ro,relatime,seclabel,devices)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (ro,relatime,seclabel,cpu,cpuacct)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (ro,relatime,seclabel,net_cls,net_prio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (ro,relatime,seclabel,perf_event)
cgroup on /sys/fs/cgroup/systemd type cgroup (ro,relatime,seclabel,xattr,name=systemd)
tmpfs on /proc/acpi type tmpfs (ro,relatime,context="system_u:object_r:container_file_t:s0:c63,c525",size=0k)
tmpfs on /proc/scsi type tmpfs (ro,relatime,context="system_u:object_r:container_file_t:s0:c63,c525",size=0k)
tmpfs on /sys/firmware type tmpfs (ro,relatime,context="system_u:object_r:container_file_t:s0:c63,c525",size=0k)
tmpfs on /sys/fs/selinux type tmpfs (ro,relatime,context="system_u:object_r:container_file_t:s0:c63,c525",size=0k)
proc on /proc/asound type proc (ro,relatime)
proc on /proc/bus type proc (ro,relatime)
proc on /proc/fs type proc (ro,relatime)
proc on /proc/irq type proc (ro,relatime)
proc on /proc/sys type proc (ro,relatime)
proc on /proc/sysrq-trigger type proc (ro,relatime)
```
versus
```
# podman run --privileged fedora mount | grep \(ro
#```

**Comment by vrothberg** [comment]
@rhatdan, I was looking at exactly that atm. CRI-O is only setting sys and cgroup ... at least in the code.

**Comment by vrothberg** [comment]
We still need to backport #2627, but the mounts are now rw on master.

**Comment by vrothberg** [comment]
Backports for:
* 1.13: https://github.com/cri-o/cri-o/pull/2639
* 1.14: https://github.com/cri-o/cri-o/pull/2635
* 1.15: https://github.com/cri-o/cri-o/pull/2634

**Comment by matthias50** [comment]
So I'm messing around with 1.14.9 and this issue is back. From spelunking in the git log, It looks like it's because there hasn't been a 1.14 release since the was merged to the 1.14 release branch. @haircommander can you say when that will happen?

**Comment by haircommander** [comment]
release 1.14.11 is out now @matthias50 sorry for the delay
